src/org/exist/storage/BrokerPool.java

```diff
@@ -1004,8 +758,9 @@ protected void initialize() throws EXistException, DatabaseConfigurationExceptio
 
                             sync(broker, Sync.MAJOR);
 
+                            //TODO(AR) remove this!
                             //require to allow access by BrokerPool.getInstance();
-                            instances.put(instanceName, this);
+                            //instances.put(instanceName, this);
```

Comment:
initialization have two phases: BrokerPool startup & subsystems startup. I'm agree with you that on BrokerPool startup phase it must not go to `instances`, but disagree that on subsystems startup phase sybsystem must not have access to `BrokerPool`. At time of that phase db is up and running.

If follow your logic linux must not startup if any subsystem fail, but it's not happening.